robot_smach_states/benchmark/benchmark_grasping.py

```diff
@@ -0,0 +1,184 @@
+                          navigation_area=inspect_from_area)
+
+        record['inspect_start'] = time.time()
+        assert inspect.execute() == 'done', "I could not inspect the support entity"
+        record['inspect_end'] = time.time()
+        record['inspect_duration'] = '{:3.3f}'.format(record['inspect_end'] - record['inspect_start'])
+
+        inspection_result = entity_ids.resolve()  # type: List[ClassificationResult]
+        if inspection_result:
+            if grasp_cls.resolve() == '*' or non_strict_class:
+                rospy.loginfo("Any item will match")
+                matching_results = inspection_result
+            else:
+                matching_results = [result for result in inspection_result if result.type == grasp_cls.resolve()]
+                rospy.loginfo("Found {} items of class {}".format(len(matching_results), grasp_cls.resolve()))
+
+            if matching_results:
+                if len(matching_results) > 1:
+                    rospy.logwarn("There are multiple items OK to grab, will select the last one")
+                    record['observed_class'] = ' '.join([result.type for result in matching_results])
+                else:
+                    record['observed_class'] = matching_results[-1].type
+                selected_entity_id = matching_results[-1].id
+
+                rospy.loginfo("Selected entity {} for grasping".format(selected_entity_id))
+                grasp_entity = ds.EdEntityDesignator(robot, id=selected_entity_id)
+                record['id'] = selected_entity_id[:6]
+
+                entity = grasp_entity.resolve()  # type: Entity
+                if entity:
+                    vector_stamped = entity.pose.extractVectorStamped()  # type: VectorStamped
+                    record['x'] = '{:.3f}'.format(vector_stamped.vector.x())
+                    record['y'] = '{:.3f}'.format(vector_stamped.vector.y())
+                    record['z'] = '{:.3f}'.format(vector_stamped.vector.z())
+
+                grab_state = Grab(robot, grasp_entity, arm)
+
+                record['grab_start'] = time.time()
+                assert grab_state.execute() == 'done', "I couldn't grasp"
+                record['grab_end'] = time.time()
+                record['grab_duration'] = '{:3.3f}'.format(record['grab_end'] - record['grab_start'])
+
+                assert nav_to_start.execute() == 'arrived', "I could not navigate back to the start"
+
+                rospy.logwarn("Robot will turn around to drop the {}".format(grasp_cls.resolve()))
+
+                force_drive = ForceDrive(robot, 0, 0, 1, 3.14)  # rotate 180 degs in pi seconds
+                force_drive.execute()
+
+                say_drop = Say(robot, sentence="I'm going to drop the item, please hold it!")
+                say_drop.execute()
+
+                drop_it = SetGripper(robot, arm_designator=arm, grab_entity_designator=grasp_entity)
```

Comment:
There is not really a reason to sent the grasp_entity, but it doesn't hurt anybody either. Up to you!

Reply:
This clear the entity from the things the arm is holding.